### PR TITLE
[DOC-189] Added backfill index flags for YB-Master and YB-TServer pages.

### PR DESCRIPTION
**Pull request by aishwarya24**
Summary -
This PR includes description for four advanced flags :
- ysql_index_backfill_rpc_timeout_ms 
- backfill_index_client_rpc_timeout_ms
- backfill_index_write_batch_size
- backfill_index_timeout_grace_margin_ms

Build URL - https://deploy-preview-20527--infallible-bardeen-164bc9.netlify.app/stable/api/ysql/the-sql-language/statements/ddl_create_index/#common-errors-and-solutions

### CODE REVIEW

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 1]
```diff
@@ -1283,6 +1283,26 @@ Valid values are `-1` (unlimited), `integer` (in kilobytes), `nMB` (in megabytes
...
+
+The time to exclude from the YB-Master flag [ysql_index_backfill_rpc_timeout_ms](../yb-master/#ysql-index-backfill-rpc-timeout-ms) in order to return results to YB-Master in the specified deadline. Should be set to at least the amount of time each batch would require, and less than `ysql_index_backfill_rpc_timeout_ms`.
+
+Default: 1 second (auto-computed)
+
+#### backfill_index_write_batch_size
+
+The number of table rows to backfill at a time. In case of [GIN indexes](../../../explore/indexes-constraints/gin/), the number can include more index rows.
+
+Default 128
```
Comment (jasonyb) [inline docs/content/preview/reference/configuration/yb-tserver.md:1304]:
```suggestion
Default: 128
```

File: docs/content/stable/reference/configuration/yb-tserver.md [hunk 2]
```diff
@@ -1283,6 +1283,26 @@ Valid values are `-1` (unlimited), `integer` (in kilobytes), `nMB` (in megabytes
...
+
+The time to exclude from the YB-Master flag [ysql_index_backfill_rpc_timeout_ms](../yb-master/#ysql-index-backfill-rpc-timeout-ms) in order to return results to YB-Master in the specified deadline. Should be set to at least the amount of time each batch would require, and less than `ysql_index_backfill_rpc_timeout_ms`.
+
+Default: 1 second (auto-computed)
+
+#### backfill_index_write_batch_size
+
+The number of table rows to backfill at a time. In case of [GIN indexes](../../../explore/indexes-constraints/gin/), the number can include more index rows.
+
+Default 128
```
Comment (jasonyb) [inline docs/content/stable/reference/configuration/yb-tserver.md:1304]:
```suggestion
Default: 128
```

**Review by jasonyb — APPROVED** [review]

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 3]
```diff
@@ -1283,6 +1283,26 @@ Valid values are `-1` (unlimited), `integer` (in kilobytes), `nMB` (in megabytes
 
 Default: `1GB`
 
+## Advanced flags
+
+#### backfill_index_client_rpc_timeout_ms
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yb-tserver.md:1288]:
```suggestion
##### backfill_index_client_rpc_timeout_ms
```

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 4]
```diff
@@ -1283,6 +1283,26 @@ Valid values are `-1` (unlimited), `integer` (in kilobytes), `nMB` (in megabytes
...
+
+Timeout (in milliseconds) for the backfill stage of a concurrent CREATE INDEX.
+
+Default: 1 day
+
+#### backfill_index_timeout_grace_margin_ms
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yb-tserver.md:1294]:
```suggestion
##### backfill_index_timeout_grace_margin_ms
```

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 5]
```diff
@@ -1283,6 +1283,26 @@ Valid values are `-1` (unlimited), `integer` (in kilobytes), `nMB` (in megabytes
...
+
+The time to exclude from the YB-Master flag [ysql_index_backfill_rpc_timeout_ms](../yb-master/#ysql-index-backfill-rpc-timeout-ms) in order to return results to YB-Master in the specified deadline. Should be set to at least the amount of time each batch would require, and less than `ysql_index_backfill_rpc_timeout_ms`.
+
+Default: 1 second (auto-computed)
+
+#### backfill_index_write_batch_size
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yb-tserver.md:1300]:
```suggestion
##### backfill_index_write_batch_size
```

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 6]
```diff
@@ -1283,6 +1283,26 @@ Valid values are `-1` (unlimited), `integer` (in kilobytes), `nMB` (in megabytes
...
+
+Timeout (in milliseconds) for the backfill stage of a concurrent CREATE INDEX.
+
+Default: 1 day
+
+#### backfill_index_timeout_grace_margin_ms
+
+The time to exclude from the YB-Master flag [ysql_index_backfill_rpc_timeout_ms](../yb-master/#ysql-index-backfill-rpc-timeout-ms) in order to return results to YB-Master in the specified deadline. Should be set to at least the amount of time each batch would require, and less than `ysql_index_backfill_rpc_timeout_ms`.
+
+Default: 1 second (auto-computed)
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yb-tserver.md:1298]: ms so 60000 no? what does auto-computed mean?

File: docs/content/preview/reference/configuration/yb-tserver.md [hunk 7]
```diff
@@ -1283,6 +1283,26 @@ Valid values are `-1` (unlimited), `integer` (in kilobytes), `nMB` (in megabytes
 
 Default: `1GB`
 
+## Advanced flags
+
+#### backfill_index_client_rpc_timeout_ms
+
+Timeout (in milliseconds) for the backfill stage of a concurrent CREATE INDEX.
+
+Default: 1 day
```
Comment (ddhodge) [inline docs/content/preview/reference/configuration/yb-tserver.md:1292]: ms, so whatever that converts to, followed by (1 day)

File: docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md [hunk 8]
```diff
@@ -302,9 +302,9 @@ DROP INDEX
 
   Do any or all of the following:
 
-  - Increase [YB-Master flag][yb-master] `ysql_index_backfill_rpc_timeout_ms` from 60000 (one minute) to 300000 (five minutes).
-  - Increase [YB-TServer flag][yb-tserver] `backfill_index_timeout_grace_margin_ms` from -1 (one second) to 60000 (one minute).
-  - Decrease [YB-TServer flag][yb-tserver] `backfill_index_write_batch_size` from 128 to 32.
+  - Increase the YB-Master flag [ysql_index_backfill_rpc_timeout_ms](../../../../../reference/configuration/yb-master/#ysql-index-backfill-rpc-timeout-ms) from 60000 (one minute) to 300000 (five minutes).
+  - Increase the YB-TServer flag [backfill_index_timeout_grace_margin_ms](../../../../../reference/configuration/yb-tserver/#backfill-index-timeout-grace-margin-ms) from -1 (one second) to 60000 (one minute).
```
[the remark this answers is not in the document]
Comment (jasonyb) [reply docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md:306]: autocomputed. you can get rid of the "from foo" to not deal with it

File: docs/content/preview/reference/configuration/yb-tserver.md (hunk 6 above)
Comment (jasonyb) [reply docs/content/preview/reference/configuration/yb-tserver.md:1298]: default is -1.  by default, that is internally computed to mean 1 second.  other flags may influence this computation, but that is complicated to explain, and this is an advanced flag anyway

Comment (ddhodge) [reply docs/content/preview/reference/configuration/yb-tserver.md]: I see. Are other flag defaults, like '1 minute' also actually autocomputed, or should they be '60000 (1 minute)'

**Pull request merged**